var/spack/repos/builtin/packages/hdf5-vol-async/package.py

```diff
@@ -35,9 +35,8 @@ class Hdf5VolAsync(CMakePackage):
     depends_on("hdf5@1.14.0: +mpi +threadsafe")
 
     # Require MPI_THREAD_MULTIPLE.
-    depends_on("openmpi +thread_multiple", when="^openmpi@:2")
-    depends_on("openmpi", when="^openmpi@3:")
-    depends_on("mvapich2 threads=multiple", when="^mvapich2")
+    depends_on("openmpi +thread_multiple", when="^[virtuals=mpi] openmpi@:2")
+    depends_on("mvapich2 threads=multiple", when="^[virtuals=mpi] mvapich2")
```

Comment:
I don't think this fixes the actual problem -- which is the semantics of ^openmpi, ^mvapich2, etc... but if this fixes CI for now I think we can merge.

Reply:
Somehow, this is also a more precise specification that we want this if we use a package to provide mpi.